lib/pry-rails/prompt.rb

```diff
@@ -22,20 +22,27 @@ def project_name
     proc do |target_self, nest_level, pry|
       "[#{pry.input_array.size}] " \
         "[#{Prompt.project_name}][#{Prompt.formatted_env}] " \
-        "#{Pry.config.prompt_name}(#{Pry.view_clip(target_self)})" \
+        "#{pry.config.prompt_name}(#{Pry.view_clip(target_self)})" \
         "#{":#{nest_level}" unless nest_level.zero?}> "
     end,
     proc do |target_self, nest_level, pry|
       "[#{pry.input_array.size}] " \
         "[#{Prompt.project_name}][#{Prompt.formatted_env}] " \
-        "#{Pry.config.prompt_name}(#{Pry.view_clip(target_self)})" \
+        "#{pry.config.prompt_name}(#{Pry.view_clip(target_self)})" \
         "#{":#{nest_level}" unless nest_level.zero?}* "
     end
   ]
 
-  Pry::Prompt::MAP["rails"] = {
-    value: RAILS_PROMPT,
-    description: "Includes the current Rails environment and project folder name.\n" \
-                 "[1] [project_name][Rails.env] pry(main)>"
-  }
+  desc = "Includes the current Rails environment and project folder name.\n" \
+          "[1] [project_name][Rails.env] pry(main)>"
+  if Pry::Prompt.respond_to?(:add)
+    Pry::Prompt.add 'rails', desc, %w(> >>) do |target, nesting, _pry_, sep|
+      sep == '>' ? RAILS_PROMPT[0].call(target, nesting, _pry_) : RAILS_PROMPT[1].call(target, nesting, _pry_)
```

Comment:
We can inline the proc and delete the `RAILS_PROMPT` constant:

```suggestion
"[#{pry.input_ring.size}] " \
  "[#{Prompt.project_name}][#{Prompt.formatted_env}] " \
  "#{pry.config.prompt_name}(#{Pry.view_clip(target_self)})" \
  "#{":#{nest_level}" unless nest_level.zero?}#{sep} "
```

Reply:
that's much better, thank you
rebased and pushed.